homeassistant/components/mqtt/tag.py

```diff
@@ -107,11 +107,11 @@ def __init__(
         config_entry: ConfigEntry,
     ) -> None:
         """Initialize."""
-        self._config = config
-        self._config_entry = config_entry
-        self.device_id = device_id
-        self.discovery_data = discovery_data
-        self.hass = hass
+        self._config: ConfigType = config
+        self._config_entry: ConfigEntry = config_entry
+        self.device_id: str | None = device_id
+        self.discovery_data: DiscoveryInfoType = discovery_data
+        self.hass: HomeAssistant = hass
```

Comment:
I think these are all inferred now
```suggestion
        self._config = config
        self._config_entry = config_entry
        self.device_id = device_id
        self.discovery_data = discovery_data
        self.hass = hass
```